packages/search-in-workspace/src/node/ripgrep-search-in-workspace-server.ts

```diff
@@ -93,47 +156,94 @@ export class RipgrepSearchInWorkspaceServer implements SearchInWorkspaceServer {
         this.client = client;
     }
 
-    protected getArgs(options?: SearchInWorkspaceOptions): string[] {
-        const args = ['--hidden', '--json'];
-        args.push(options && options.matchCase ? '--case-sensitive' : '--ignore-case');
+    protected getArgs(rootPaths: string[], options?: SearchInWorkspaceOptions): string[] {
+        const args = new Set<string>();
+
+        const appendGlobArgs = (rawPatterns: string[], exclude: boolean) => {
+            rootPaths.forEach(root => {
+                for (const rawPattern of rawPatterns) {
+                    if (rawPattern !== '') {
+                        const globArguments = this.filePatternToGlobs(rawPattern, root, exclude);
+                        globArguments.forEach(arg => args.add(arg));
+                    }
+                }
+            });
+        };
+
+        args.add('--hidden');
+        args.add('--json');
+
+        args.add(options && options.matchCase ? '--case-sensitive' : '--ignore-case');
         if (options && options.includeIgnored) {
-            args.push('--no-ignore');
+            args.add('--no-ignore');
         }
         if (options && options.maxFileSize) {
-            args.push('--max-filesize=' + options.maxFileSize.trim());
+            args.add('--max-filesize=' + options.maxFileSize.trim());
         } else {
-            args.push('--max-filesize=20M');
+            args.add('--max-filesize=20M');
         }
+
         if (options && options.include) {
-            for (const include of options.include) {
-                if (include !== '') {
-                    args.push('--glob=**/' + include);
-                }
-            }
+            appendGlobArgs(options.include, false);
         }
+
         if (options && options.exclude) {
-            for (const exclude of options.exclude) {
-                if (exclude !== '') {
-                    args.push('--glob=!**/' + exclude);
-                }
-            }
+            appendGlobArgs(options.exclude, true);
         }
+
         if (options && options.useRegExp || options && options.matchWholeWord) {
-            args.push('--regexp');
+            args.add('--regexp');
         } else {
-            args.push('--fixed-strings');
-            args.push('--');
+            args.add('--fixed-strings');
+            args.add('--');
         }
-        return args;
+
+        return Array.from(args);
     }
 
+    /**
+     * Transforms a file pattern to a glob pattern (unless it's already a glob).
+     * It then creates corresponding 'ripgrep' CLI parameters,
```

Comment:
It's possible the method does too much, it might be one method to transform the pattern to a glob.
Converting it to CLI arguments seems unrelated.

If that is the intention of the method, it might be better to name it differently.

Reply:
Updated the jsdoc and method name to: patternToGlobCLIArguments
this better reflects the purpose of the method.  Thanks!